Sources/NIO/CircularBuffer.swift

```diff
@@ -33,6 +33,30 @@ public struct CircularBuffer<E>: CustomStringConvertible, AppendableCollection {
     /// The index into the buffer of the next free slot
     private(set) /* private but tests */ internal var tailIdx = 0
 
+    public struct Index: Strideable {
+        @usableFromInline
+        var _value: Int
+
+        public typealias Stride = Int
+
+        @inlinable
+        internal init(_value: Int) {
+            self._value = _value
+        }
+
+        @inlinable
+        public func distance(to other: Index) -> Int {
+            return self._value.distance(to: other._value)
+        }
+
+        @inlinable
+        public func advanced(by n: Int) -> Index {
+            return Index(_value: self._value + n)
+        }
+    }
```

Comment:
This is a note for the future, we may find it useful to implement the `Comparable` functions ourselves rather than use the synthesised implementations on `Strideable`, just to avoid stressing the inliner too badly.

Reply:
done too :)